File: IntersectionOfTwoLinkedList.cpp
#include <iostream>

struct ListNode {
    int val;
    ListNode* next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
    ListNode* getIntersectionNode(ListNode* headA, ListNode* headB) {
        ListNode* cur = headA;
        ListNode* curO = headB;

        while (cur != curO) {
            cur = cur == nullptr ? headB : cur->next;
            curO = curO == nullptr ? headA : curO->next;
        }

        return cur;
    }
};

File: IntersectionOfTwoLinkedList.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def getIntersectionNode(self, headA, headB):
        cur = headA
        curO = headB
        
        while cur != curO:
            cur = headB if cur is None else cur.next
            curO = headA if curO is None else curO.next
            
        return cur

